interactions/client/bot.py

```diff
@@ -363,6 +365,55 @@ async def change_presence(self, presence: ClientPresence) -> None:
         """
         await self._websocket._update_presence(presence)
 
+    async def get(
+        self, model: T, http_method: Optional[str] = None, **kwargs: Dict[str, Snowflake]
+    ) -> T:
+        r"""
+        A helper function for getting a model from the Discord API.
+
+        :param model: The model to get.
+        :type model: object
+        :param http_method: The HTTP method to use instead of `get_x`.
+        :type http_method: Optional[str]
+        :param \**kwargs: The arguments to pass to the HTTP method.
+        :type \**kwargs: Dict[str, Snowflake]
+        :return: The model.
+        :rtype: object
+        """
+        if model is Role:
+            if not kwargs.get("guild_id") and kwargs.get("role_id"):
+                raise TypeError(
+                    "Client.get() missing some of the following required keyword arguments: guild_id, role_id."
+                )
+            try:
+                roles: List[dict] = await self._http.get_all_roles(int(kwargs["guild_id"]))
+            except TypeError as e:
+                raise ValueError("Invalid guild_id provided.") from e
+            role = next((_role for _role in roles if _role["id"] == str(kwargs["role_id"])), None)
+            if role is None:
+                raise ValueError("Role not found.")
+            return Role(**role, _client=self._http)
+        else:
+            class_name: str = model.__name__.removesuffix("s")
+            try:
+                _http_method: Coroutine = getattr(
+                    self._http, http_method or f"get_{class_name.lower()}"
+                )
+            except AttributeError as e:
+                raise ValueError(f"Client.get() does not support the model {class_name}.") from e
+            try:
+                res: Union[dict, List[dict]] = await _http_method(**kwargs)
+            except TypeError as e:
+                raise ValueError(
+                    f"Client.get() could not find a {class_name} with the given IDs."
+                ) from e
```

Comment:
```suggestion
                )
```

Please raise only the exception and none more. Developers should be able to catch for either exception and not clump it as one. If this isn't possible, please make use of our current exception system, which I will note is bound to be refactored.

Reply:
Removed them